### DISCUSSION TITLE
Porting protobufc onto ARM TEE platform

### DISCUSSION THREAD

**Discussion by haleyyew** [protobuf-c/protobuf-c#648] [Q&A] [answered]
Hi,
I am trying to use protobuf on ARM TrustZone. The problem is that the libc that the ARM build environment uses does not have some of the functions that the standard libc has (e.g., malloc, memcpy, memmove). In my environment, these functions exist in a separate static library called libutil.a. How do I compile protobuf so that rather than looking for these functions dynamically in libc, it'll look for them statically in libutil?

Thanks.

**Comment by edmonds** [comment, marked as answer]
protobuf-c uses the GNU Build System. If you run `./configure --help` it will print out a usage summary that includes environment variables that can be set to override the compiler flags, linker flags, etc.

**Comment by haleyyew** [comment]
Can you give me more tips on how to replace libc with libutil.a when compiling protobuf? I'm not quite sure which linker flags you are referring to.

Thanks.
